src/features/More/components/categoryItem.tsx

```diff
@@ -11,52 +13,122 @@ interface CategoryItemProps {
 }
 
 const CategoryItem = (props: CategoryItemProps) => {
+  const [subCategoriesHeight, setSubCategoriesHeight] = useState<
+    number | undefined
+  >(0);
+
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const handlePress = () => {
+    props.setIndexOfPressedCategory(props.itemData.index);
+    setIsExpanded(!isExpanded);
+  };
+
+  useEffect(() => {
+    if (props.isSelected) {
+      if (isExpanded) {
+        setSubCategoriesHeight(undefined);
+      } else {
+        setSubCategoriesHeight(0);
+      }
+    } else {
+      setSubCategoriesHeight(0);
+      setIsExpanded(false);
+    }
+  }, [isExpanded, props.isSelected]);
+
   return (
-    <TouchableOpacity
-      style={{
-        ...styles.listItem,
-        backgroundColor: props.isSelected ? Colors.ACTIVE_ICON : Colors.WHITE,
-      }}
-      onPress={() => {
-        props.setIndexOfPressedCategory(props.itemData.index);
-      }}
-    >
-      <View style={styles.itemContainer}>
-        <View style={styles.arrowIcon}>
-          <ArrowIcon />
-        </View>
-        <View>
-          <Text style={styles.title}>{props.itemData.item?.title}</Text>
+    <View>
+      <TouchableOpacity
+        style={{
+          ...styles(subCategoriesHeight).listItem,
+          backgroundColor: props.isSelected ? Colors.ACTIVE_ICON : Colors.WHITE,
+        }}
+        onPress={handlePress}
+      >
+        <View style={styles(subCategoriesHeight).itemContainer}>
+          {props.itemData.item.children.length !== 0 && (
+            <View style={styles(subCategoriesHeight).arrowIcon}>
+              <ArrowIcon
+                transform={
+                  isExpanded
+                    ? Strings.moreScreen.UP_ARROW
+                    : Strings.moreScreen.DOWN_ARROW
+                }
+              />
+            </View>
+          )}
+
+          <View style={styles(subCategoriesHeight).titleContainer}>
+            <Text style={styles(subCategoriesHeight).title}>
+              {props.itemData.item?.title}
+            </Text>
+          </View>
         </View>
+      </TouchableOpacity>
+
+      <View style={styles(subCategoriesHeight).subCategories}>
```

Comment:
This should be a conditional rendering and not always render a view with height 0.

Reply:
Fixed. Could you check?

Reply:
yes